### PR TITLE
Select: add support for typeahead select

### PR DESCRIPTION
**Pull request by jrusz**
When there is an item selected it would click on the clear button instead of the toggle button when trying to access items or calling open(), this fixes that.

### CODE REVIEW

**Comment by jrusz** [comment]
@digitronik Hi, can you review please? 

Also there were some seemingly random failures during the test runs, I had to force push a few times before it all passed, they were tests for other widgets, unrelated to my change though, should we be concerned about that?

**Comment by jrusz** [comment]
I've actually found some issues with this. Need to update it.

**Comment by jrusz** [comment]
This should work now, also I've found out that I need it for the `fill` method, it doesn't work without this. I think it's the "Typeahead" select which causes these issues for me with a default value preselected (but the same would happen if you were to select something twice) ... I'll add some tests for this as it seems that select type is not covered.

**Comment by jrusz** [comment]
I ended up adding a new class for the select as the `read` method didn't work, the selected value is a in separate `input` element instead underneath the button, the rest seems to be fine.

File: src/widgetastic_patternfly5/components/menus/select.py [hunk 1]
```diff
@@ -182,3 +187,14 @@ def items(self):
 
 class CheckboxSelect(BaseCheckboxSelect, Dropdown):
     DEFAULT_LOCATOR = './/div[contains(@class, "-c-select")][1]'
+
+
+class BaseTypeaheadSelect(BaseSelect):
+    text = TextInput(locator=".//input")
+
+    def read(self):
+        return self.browser.get_attribute("value", self.text)
```
Comment (digitronik) [inline src/widgetastic_patternfly5/components/menus/select.py:196]: Lets make this part of widget directly only single method.
Better to use `input` as name of child widget. 
```
input = TextInput(locator=".//input")
```

File: src/widgetastic_patternfly5/components/menus/select.py [hunk 2]
```diff
@@ -19,7 +20,11 @@ class BaseSelect:
     https://www.patternfly.org/components/menus/select
     """
 
-    BUTTON_LOCATOR = ".//button"
+    BUTTON_LOCATOR = (
```
Comment (digitronik) [inline src/widgetastic_patternfly5/components/menus/select.py:23]: Better to overwrite in widget itself. Those limitation can break usage.  Not tested against console dot so.

**Review by digitronik — CHANGES_REQUESTED** [review]
Nice PR :+1: 

Some suggestions
As we are introducing new widget better to maintain it in README as well. 

Special thanks for covering unit tests.  I will take a look on more time after changes.

**Comment by jrusz** [comment]
@digitronik Is this what you meant?

**Comment by jrusz** [comment]
I'm not sure why, but the `ouia` component doesn't work if I don't use the Base component type, is it ok like this?

**Review by digitronik — APPROVED** [review]

**Comment by digitronik** [comment]
@jrusz I'm good with it can you please check jobs

**Comment by jrusz** [comment]
> @jrusz I'm good with it can you please check jobs

Thank you. I don't know why the jobs fail, like I said in a comment above they fail seemingly randomly. I don't have permissions to restart them, I'd have to force push, can you try restarting them please?

File: src/widgetastic_patternfly5/components/menus/select.py [hunk 3]
```diff
@@ -182,3 +183,19 @@ def items(self):
 
 class CheckboxSelect(BaseCheckboxSelect, Dropdown):
     DEFAULT_LOCATOR = './/div[contains(@class, "-c-select")][1]'
+
+
+class BaseTypeaheadSelect(BaseSelect):
+    BUTTON_LOCATOR = (
+        ".//button[(contains(@class, '-c-select__toggle') "
+        "or contains(@class, '-c-menu-toggle')) "
+        "and not(contains(@class, '-c-select__toggle-clear'))]"
```
Comment (mshriver) [inline src/widgetastic_patternfly5/components/menus/select.py:192]: Consider creating a separate locator for the clear button, and a function to click it.

**Review by mshriver — APPROVED** [review]

**Pull request merged**